File: compute_science/bubbleSort.js
function swap(smallIndex, largeIndex, arr) {
  const temp = arr[smallIndex];
  arr[smallIndex] = arr[largeIndex];
  arr[largeIndex] = temp;
}

let count = 0;
function bubbleSort(arr) {
  let noSwap = true;
  for (let i = arr.length - 1; i >= 0; i--) {
    noSwap = true;
    for (let j = 0; j <= i - 1; j++) {
      count++;
      if (arr[j + 1] < arr[j]) {
        swap(j, j + 1, arr);
        noSwap = false;
      }
    }
    if (noSwap) break;
  }
}

const testArr = [3, 7, 1, 6, 3, 5, 123, 234, 312435, 356];
bubbleSort(testArr);
console.log(testArr);
console.log("Count: ", count);

File: compute_science/bubble_sort.py
def swap(left, right, arr):
    temp = arr[left]
    arr[left] = arr[right]
    arr[right] = temp


# Todo: add isSwap optimization
def bubble_sort(arr):
    for i in range(0, len(arr)):
        swapped = False
        for j in range(0, len(arr) - i - 1):
            if arr[j] > arr[j + 1]:
                swap(j, j + 1, arr)
                swapped = True
        if not swapped:
            break

    return arr


testArr = [64, 34, 25, 12, 22, 11, 90]
print(bubble_sort(testArr))
